nft_ingester/src/parsers/bubblegum.rs

```diff
@@ -232,21 +267,23 @@ async fn handle_bubblegum_instruction<'a, 'b, 't>(
             let leaf_event = get_leaf_event(logs)?;
             db.transaction::<_, _, IngesterError>(|txn| {
                 Box::pin(async move {
-                    save_changelog_events(gummy_roll_events, slot, txn).await?;
+                    let seq = save_changelog_events(gummy_roll_events, slot, txn)
+                        .await?
+                        .ok_or(IngesterError::ChangeLogEventMalformed)?;
                     match leaf_event.schema {
                         LeafSchema::V1 {
                             id,
-                            delegate,
                             ..
                         } => {
-                            let _delegate_bytes = delegate.to_bytes().to_vec();
                             let id_bytes = id.to_bytes().to_vec();
                             let asset_to_update = asset::ActiveModel {
                                 id: Unchanged(id_bytes.clone()),
                                 burnt: Set(true),
                                 ..Default::default()
                             };
-                            update_asset(txn, id_bytes, asset_to_update).await
+                            // Don't send sequence number with this update, because we will always
+                            // run this update even if it's from a backfill/replay.
+                            update_asset(txn, id_bytes, None, asset_to_update).await
```

Comment:
It would be nice if we could use `seq` with every Bubblegum instruction.  However, in order to do this, every Bubblegum instruction would need to provide all the same fields.  As an example, `burnt` is not part of the on-chain leaf schema.  On-chain, the leaf is zeroed out, which is the same as with `redeem` except for the voucher.  Thus, if we missed a `burn` instruction and skipped writing the `burnt` flag due to `seq` number, we would not know if our off-chain asset was burned or redeemed.

So we need to actually always replay the `burn` instruction to make sure we set the `burnt` flag, and that only works because `burn` is orthogonal to the other instructions that modify `id`, `leaf`, `delegate`, and `owner`.

### Overall
In general, to make using a sequence number for the whole table work, there are two safe solutions:
1. Instructions must update all the same fields (`id`, `leaf`, `delegate`, `owner`).  Then if the table already has a greater `seq` number, we know the table already has newer data for those same fields (`id`, `leaf`, `delegate`, `owner`) so we don't rerun the instruction and risk putting in older stale data.
2. Instructions must only update unique fields.  `Burn` only updates a 'burnt' flag and nobody else does.  `Burn` also does not update `id`, `leaf`, `delegate`, or `owner`.  So `burn` can be rerun out of order and won't overwrite old data to anything (unless there was an `unburn` which doesn't make sense).

### Mint and decompress
You will note that both `mint` and `decompress` don't actually follow the two rules above, so then we have to just reason about them.

#### Minting
Any instructions after `mint` that change `id`, `leaf`, `delegate`, or `owner` will attempt to do an `UPDATE` to the existing `asset` table entry.  So they will actually fail if we haven't done the initial asset creation for the `mint` instruction processing.

Also, if the entry already exists and and we were to replay the `mint` without checking `seq`, we would potentially overwrite old data for `id`, `leaf`, `delegate`, or `owner`.

We could check `seq` and only rerun the `mint` instruction if we have a newer `seq` number, but that doesn't really make sense because we shouldn't ever be seeing a newer `mint` instruction for the same asset - there will only be one `mint` for an asset.

#### Decompress
After an asset is decompressed it is no longer managed by Bubblegum and Gummyroll so there can no longer be Bubblegum/Gummyroll events after this.

Also, `seq` is not part of the decompress event that is picked up by plerkle.  So yes this instruction modifies the `id` and `leaf` but it is currently a one-way door for the asset so if we missed it, we would rerun it because nothing will come after this.

Note this assumption falls apart if we add `compress` or other Bubblegum instructions that could occur after an asset was decompressed.

### Summary
I think what is in this PR will currently work with backfilling/replaying old transactions.  But it seems very fragile and not extensible, and it would be better to have some kind of versioning on every data field in the asset and related tables.